[2024-06-20T11:43:14.359+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-06-20T11:43:14.376+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: dag_data_etl_v30.data_etl.preprocess_data.preprocess_data scheduled__2024-06-19T12:00:00+00:00 [queued]>
[2024-06-20T11:43:14.382+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: dag_data_etl_v30.data_etl.preprocess_data.preprocess_data scheduled__2024-06-19T12:00:00+00:00 [queued]>
[2024-06-20T11:43:14.385+0000] {taskinstance.py:2306} INFO - Starting attempt 2 of 6
[2024-06-20T11:43:14.398+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): data_etl.preprocess_data.preprocess_data> on 2024-06-19 12:00:00+00:00
[2024-06-20T11:43:14.406+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'dag_data_etl_v30', 'data_etl.preprocess_data.preprocess_data', 'scheduled__2024-06-19T12:00:00+00:00', '--job-id', '141', '--raw', '--subdir', 'DAGS_FOLDER/data_etl_dag.py', '--cfg-path', '/tmp/tmp3o_8l0_p']
[2024-06-20T11:43:14.408+0000] {standard_task_runner.py:91} INFO - Job 141: Subtask data_etl.preprocess_data.preprocess_data
[2024-06-20T11:43:14.408+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=1638) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-06-20T11:43:14.409+0000] {standard_task_runner.py:63} INFO - Started process 1642 to run task
[2024-06-20T11:43:14.441+0000] {task_command.py:426} INFO - Running <TaskInstance: dag_data_etl_v30.data_etl.preprocess_data.preprocess_data scheduled__2024-06-19T12:00:00+00:00 [running]> on host 10482076056f
[2024-06-20T11:43:14.499+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Solarev Ruslan' AIRFLOW_CTX_DAG_ID='dag_data_etl_v30' AIRFLOW_CTX_TASK_ID='data_etl.preprocess_data.preprocess_data' AIRFLOW_CTX_EXECUTION_DATE='2024-06-19T12:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-06-19T12:00:00+00:00'
[2024-06-20T11:43:14.501+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-06-20T11:43:14.502+0000] {python.py:237} INFO - Done. Returned value was: {{ task_instance.xcom_pull(task_ids='preprocess_tags', dag_id='adhoc_airflow', key='return_value') }}
[2024-06-20T11:43:14.502+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-06-20T11:43:14.508+0000] {xcom.py:675} ERROR - Object of type PlainXComArg is not JSON serializable. If you are using pickle instead of JSON for XCom, then you need to enable pickle support for XCom in your *** config or make sure to decorate your object with attr.
[2024-06-20T11:43:14.509+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/json.py", line 91, in default
    return serialize(o)
           ^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/serialization/serde.py", line 189, in serialize
    raise TypeError(f"cannot serialize object of type {cls}")
TypeError: cannot serialize object of type <class '***.models.xcom_arg.PlainXComArg'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 486, in _execute_task
    task_instance.xcom_push(key=XCOM_RETURN_KEY, value=xcom_value, session=session)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 76, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3197, in xcom_push
    XCom.set(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 76, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/xcom.py", line 246, in set
    value = cls.serialize_value(
            ^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/xcom.py", line 673, in serialize_value
    return json.dumps(value, cls=XComEncoder).encode("UTF-8")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
          ^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/json.py", line 104, in encode
    return super().encode(o)
           ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 200, in encode
    chunks = self.iterencode(o, _one_shot=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 258, in iterencode
    return _iterencode(o, 0)
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/json.py", line 93, in default
    return super().default(o)
           ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 180, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type PlainXComArg is not JSON serializable
[2024-06-20T11:43:14.514+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=dag_data_etl_v30, task_id=data_etl.preprocess_data.preprocess_data, run_id=scheduled__2024-06-19T12:00:00+00:00, execution_date=20240619T120000, start_date=20240620T114314, end_date=20240620T114314
[2024-06-20T11:43:14.522+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 141 for task data_etl.preprocess_data.preprocess_data (Object of type PlainXComArg is not JSON serializable; 1642)
[2024-06-20T11:43:14.542+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-06-20T11:43:14.550+0000] {taskinstance.py:3498} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-06-20T11:43:14.554+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
